azure/network.tf

```diff
@@ -209,7 +209,7 @@ resource "azurerm_public_ip" "redpanda" {
   resource_group_name = azurerm_resource_group.redpanda.name
   location            = azurerm_resource_group.redpanda.location
   allocation_method   = "Static"
-  availability_zone   = try(var.zone, "Zone-Redundant")
+  availability_zone   = try(var.availability_zone[count.index % length(var.availability_zone)], "Zone-Redundant")
```

Comment:
Just to make sure I understand what you're doing here:

for everything that has a count based on, like var.client_vm_instances, we're setting the az for that resource instance to whatever az is in the az list based on `Index MOD  az array count`?

So if we have a 3 node cluster, and 3 azs, we get things evenly spread. But if we have a 3 node cluster and 2 azs, we end up with a SPOF on one az because it can contain two brokers. 

We should probably be warning on this somehow.

Reply:
Your commentary is correct. `Index mod AZ Array Count` is how we're doing things. I'm not sure we need to comment this specifically - hopefully it's obvious that if you put 3 things in 2 places then they aren't evenly split